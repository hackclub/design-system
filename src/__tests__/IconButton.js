/* eslint-env jest */

import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import { IconButton } from '..'

describe('IconButton', () => {
  test('renders', () => {
    const json = renderer.create(<IconButton />).toJSON()
    expect(json).toMatchSnapshot()
  })

  test('renders with circle prop', () => {
    const json = renderer.create(<IconButton circle />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('padding', '12px')
  })

  test('executes onClick prop on click', () => {
    const handleClick = jest.fn()
    const wrapper = shallow(<IconButton onClick={handleClick} />)
    wrapper.simulate('click')
    expect(handleClick).toBeCalled()
  })
})
