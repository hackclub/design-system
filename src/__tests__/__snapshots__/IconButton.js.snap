// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton renders 1`] = `
.c1 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
}

.c0 {
  margin: 0px;
  padding-left: 18px;
  padding-right: 18px;
  padding-top: 12px;
  padding-bottom: 12px;
  background-color: transparent;
  color: #ffffff;
  font-size: 24px;
  -webkit-font-smoothing: antialiased;
  display: inline-block;
  vertical-align: middle;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  font-family: inherit;
  font-weight: bold;
  line-height: 1.125;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  cursor: pointer;
  -webkit-transition: 0.1875s cubic-bezier(0.375,0,0.675,1) box-shadow;
  transition: 0.1875s cubic-bezier(0.375,0,0.675,1) box-shadow;
  box-shadow: 0 2px 4px rgba(0,0,0,0.125);
  border-radius: 9999px;
  border: none;
  box-shadow: none !important;
  line-height: 0 !important;
}

.c0:hover,
.c0:focus {
  outline: 0;
  box-shadow: 0 2px 6px rgba(0,0,0,0.125);
}

.c0:active {
  outline: 0;
  box-shadow: 0 2px 8px 2px rgba(0,0,0,0.125);
}

<a
  className="c0"
  color="white"
  fontSize={3}
>
  <svg
    aria-label="flag"
    className="c1"
    clipRule="evenodd"
    fill="currentColor"
    fillRule="evenodd"
    height={24}
    preserveAspectRatio="xMidYMid meet"
    strokeLinejoin="round"
    strokeMiterlimit="1.414"
    viewBox="0 0 32 32"
    width={24}
    xmlns="http://www.w3.org/2000/svg"
  >
    <g>
      <path
        d="M10.953 5.034a1 1 0 0 0-1.225.707L4.034 26.992a1 1 0 1 0 1.932.517l5.694-21.25a1 1 0 0 0-.707-1.225zm2.107 9.005c.425-1.703.798-3.036 1.225-4.079.429-1.058.766-1.43.912-1.532a.216.216 0 0 0 .022-.023l.017.003c.131-.022.133-.021.353.073l.065.028c.584.23 1.492.826 2.826 2.076 1.584 1.462 3.173 2.338 4.36 2.738a9.906 9.906 0 0 0 2.045.4c-.312 1.161-.627 2.297-1.028 3.334-.405 1.061-.756 1.774-1.284 2.307-.385.41-.719.542-1.131.527-.519-.018-1.447-.289-2.901-1.37-1.746-1.291-3.25-2.073-4.327-2.514a17.61 17.61 0 0 0-1.498-.524c.08-.375.193-.838.344-1.444zm12.104-1.615a.522.522 0 0 1 0 0zm-13.21 2.816l.017.008a.08.08 0 0 1-.017-.008zm-.834-1.685c1.727-6.93 3.174-9.634 8.727-4.43 2.833 2.655 4.933 2.646 6.14 2.641 1.16-.005 1.494-.007.86 2.359-1.294 4.83-3.053 10.796-9.5 6-2.638-1.962-4.392-2.486-5.449-2.801-1.526-.456-1.599-.478-.778-3.769z"
      />
    </g>
  </svg>
</a>
`;

exports[`IconButton renders with circle prop 1`] = `
.c1 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
}

.c0 {
  margin: 0px;
  padding-left: 18px;
  padding-right: 18px;
  padding-top: 12px;
  padding-bottom: 12px;
  background-color: transparent;
  color: #ffffff;
  font-size: 24px;
  -webkit-font-smoothing: antialiased;
  display: inline-block;
  vertical-align: middle;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  font-family: inherit;
  font-weight: bold;
  line-height: 1.125;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  cursor: pointer;
  -webkit-transition: 0.1875s cubic-bezier(0.375,0,0.675,1) box-shadow;
  transition: 0.1875s cubic-bezier(0.375,0,0.675,1) box-shadow;
  box-shadow: 0 2px 4px rgba(0,0,0,0.125);
  border-radius: 9999px;
  border: none;
  box-shadow: none !important;
  line-height: 0 !important;
  padding: 12px;
}

.c0:hover,
.c0:focus {
  outline: 0;
  box-shadow: 0 2px 6px rgba(0,0,0,0.125);
}

.c0:active {
  outline: 0;
  box-shadow: 0 2px 8px 2px rgba(0,0,0,0.125);
}

<a
  className="c0"
  color="white"
  fontSize={3}
>
  <svg
    aria-label="flag"
    className="c1"
    clipRule="evenodd"
    fill="currentColor"
    fillRule="evenodd"
    height={24}
    preserveAspectRatio="xMidYMid meet"
    strokeLinejoin="round"
    strokeMiterlimit="1.414"
    viewBox="0 0 32 32"
    width={24}
    xmlns="http://www.w3.org/2000/svg"
  >
    <g>
      <path
        d="M10.953 5.034a1 1 0 0 0-1.225.707L4.034 26.992a1 1 0 1 0 1.932.517l5.694-21.25a1 1 0 0 0-.707-1.225zm2.107 9.005c.425-1.703.798-3.036 1.225-4.079.429-1.058.766-1.43.912-1.532a.216.216 0 0 0 .022-.023l.017.003c.131-.022.133-.021.353.073l.065.028c.584.23 1.492.826 2.826 2.076 1.584 1.462 3.173 2.338 4.36 2.738a9.906 9.906 0 0 0 2.045.4c-.312 1.161-.627 2.297-1.028 3.334-.405 1.061-.756 1.774-1.284 2.307-.385.41-.719.542-1.131.527-.519-.018-1.447-.289-2.901-1.37-1.746-1.291-3.25-2.073-4.327-2.514a17.61 17.61 0 0 0-1.498-.524c.08-.375.193-.838.344-1.444zm12.104-1.615a.522.522 0 0 1 0 0zm-13.21 2.816l.017.008a.08.08 0 0 1-.017-.008zm-.834-1.685c1.727-6.93 3.174-9.634 8.727-4.43 2.833 2.655 4.933 2.646 6.14 2.641 1.16-.005 1.494-.007.86 2.359-1.294 4.83-3.053 10.796-9.5 6-2.638-1.962-4.392-2.486-5.449-2.801-1.526-.456-1.599-.478-.778-3.769z"
      />
    </g>
  </svg>
</a>
`;
